rust/lance/src/io/exec/knn.rs

```diff
@@ -221,6 +221,7 @@ impl ExecutionPlan for KNNVectorDistanceExec {
         context: Arc<datafusion::execution::context::TaskContext>,
     ) -> DataFusionResult<SendableRecordBatchStream> {
         let input_stream = self.input.execute(partition, context)?;
+        let baseline_metrics = BaselineMetrics::new(&self.metrics, partition);
```

Comment:
`BaselineMetrics` this class is a bit redundant. It can use like `MetricBuilder::new(&self.metrics).elapsed_compute(partition);` to only calculate the elapsed time here